astarte/device/device.py

```diff
@@ -273,8 +273,9 @@ def send(self, interface_name: str, interface_path: str, payload: object,
         if isinstance(payload, collections.abc.Mapping):
             raise TypeError('Payload for individual interfaces should not be a dictionary')
 
-        if not self.__validate_data(interface_name, interface_path, payload, timestamp):
-            print("validation failed")
+        validate_data_res = self.__validate_data(interface_name, interface_path, payload, timestamp)
```

Comment:
```suggestion
        (validation_success, validation_error_message) = self.__validate_data(interface_name, interface_path, payload, timestamp)
```
Usually unpacking tuples results in a more readable code